pkg/objstore/s3/s3.go

```diff
@@ -47,46 +48,64 @@ type Bucket struct {
 type Config struct {
 	Bucket       string
 	Endpoint     string
-	AccessKey    string
-	SecretKey    string
+	Credentials  *credentials.Credentials
 	Insecure     bool
 	SignatureV2  bool
 	SSEEnprytion bool
 }
 
 // RegisterS3Params registers the s3 flags and returns an initialized Config struct.
 func RegisterS3Params(cmd *kingpin.CmdClause) *Config {
-	var s3config Config
+
+	var (
+		s3Config  Config
+		accessKey string
+		secretKey string
+	)
 
 	cmd.Flag("s3.bucket", "S3-Compatible API bucket name for stored blocks.").
-		PlaceHolder("<bucket>").Envar("S3_BUCKET").StringVar(&s3config.Bucket)
+		PlaceHolder("<bucket>").Envar("S3_BUCKET").StringVar(&s3Config.Bucket)
 
 	cmd.Flag("s3.endpoint", "S3-Compatible API endpoint for stored blocks.").
-		PlaceHolder("<api-url>").Envar("S3_ENDPOINT").StringVar(&s3config.Endpoint)
+		PlaceHolder("<api-url>").Envar("S3_ENDPOINT").StringVar(&s3Config.Endpoint)
 
 	cmd.Flag("s3.access-key", "Access key for an S3-Compatible API.").
-		PlaceHolder("<key>").Envar("S3_ACCESS_KEY").StringVar(&s3config.AccessKey)
+		PlaceHolder("<access-key>").StringVar(&accessKey)
 
-	s3config.SecretKey = os.Getenv("S3_SECRET_KEY")
+	cmd.Flag("s3.secret-key", "Secrety key for an S3-Compatible API.").
```

Comment:
Hmm there was discussion previously about this, that secret key should never be passed through flag. 

See  this: https://github.com/improbable-eng/thanos/pull/230

Also any reason why this and above does not have envvar defined?

Reply:
That's understandable.

I separated the envvar loading by utilizing `credentials.EnvAWS` on line 90. It will search for environmental variables.

Reply:
sure, but should we expose flag as well? When it is unsafe to specify secret via flag?

Reply:
You did mention that you wanted an option if someone was unable to modify environmental variable. I'm fine either way.

Reply:
we explicitly decided to not give that option.